core/src/test/java/ai/timefold/solver/core/impl/constructionheuristic/DefaultConstructionHeuristicPhaseTest.java

```diff
@@ -341,4 +358,164 @@ void constructionHeuristicAllocateToValueFromQueue() {
+                .withEntityClass(TestdataListMultiEntityFirstEntity.class);
+        var entityPlacerConfig = new QueuedEntityPlacerConfig();
+
+        var solverConfig = PlannerTestUtils.buildSolverConfig(
+                TestdataListMultiEntitySolution.class, TestdataListMultiEntityFirstEntity.class,
+                TestdataListMultiEntitySecondEntity.class)
+                .withPhases(new ConstructionHeuristicPhaseConfig()
+                        .withEntityPlacerConfigList(valuePlacerConfig, entityPlacerConfig)
+                        .withTerminationConfig(new TerminationConfig().withStepCountLimit(16)))
+                .withEasyScoreCalculatorClass(TestdataListMultiEntityEasyScoreCalculator.class);
+
+        var problem = TestdataListMultiEntitySolution.generateUninitializedSolution(2, 2, 2);
+        var solution = PlannerTestUtils.solve(solverConfig, problem);
+        assertThat(solution.getEntityList().stream()
+                .filter(e -> e.getValueList().isEmpty()))
+                .isEmpty();
+        assertThat(solution.getOtherEntityList().stream()
+                .filter(e -> e.getBasicValue() == null || e.getSecondBasicValue() == null))
+                .isEmpty();
+    }
+
+    @Test
+    void solveCustomConfigurationWithListAndBasicVariables() {
+        var valueSelectorConfig = new ValueSelectorConfig("valueList")
+                .withId("valueList");
+        var mimicReplayingValueSelectorConfig = new ValueSelectorConfig()
+                .withMimicSelectorRef("valueList")
+                .withVariableName("valueList");
+        var valuePlacerConfig = new QueuedValuePlacerConfig()
+                .withValueSelectorConfig(valueSelectorConfig)
+                .withMoveSelectorConfig(new ListChangeMoveSelectorConfig()
+                        .withValueSelectorConfig(mimicReplayingValueSelectorConfig));
+        var entityPlacerConfig = new QueuedEntityPlacerConfig();
+
+        var solverConfig = PlannerTestUtils.buildSolverConfig(
+                TestdataListMultiVarSolution.class, TestdataListMultiVarEntity.class, TestdataListMultiVarValue.class,
+                TestdataListMultiVarOtherValue.class)
+                .withPhases(new ConstructionHeuristicPhaseConfig()
+                        .withEntityPlacerConfigList(valuePlacerConfig, entityPlacerConfig)
+                        .withTerminationConfig(new TerminationConfig().withStepCountLimit(16)))
+                .withEasyScoreCalculatorClass(TestdataListMultiVarEasyScoreCalculator.class);
+
+        var problem = TestdataListMultiVarSolution.generateUninitializedSolution(2, 2, 2);
+        var solution = PlannerTestUtils.solve(solverConfig, problem);
+        assertThat(solution.getEntityList().stream()
+                .filter(e -> e.getBasicValue() == null || e.getSecondBasicValue() == null || e.getValueList().isEmpty()))
+                .isEmpty();
+    }
+
+    @Test
+    void solveCustomConfigurationWithListVariables() {
```

Comment:
I'm missing coverage for unassigned values, both basic and list.
I'm missing coverage for pinning.
Both have broken CH in the past, I'd be surprised if they didn't do it again this time.